packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,250 @@
+    if (claimParameters.erc20 != claim.erc20) revert IGachaLazyClaim.CannotChangePaymentToken();
+
+    // Overwrite the existing values for startDate, endDate, and cost
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claim.storageProtocol,
+      total: claim.total,
+      totalMax: claim.totalMax,
+      startDate: claimParameters.startDate,
+      endDate: claimParameters.endDate,
+      startingTokenId: claim.startingTokenId,
+      itemVariations: claim.itemVariations,
+      location: claim.location,
+      paymentReceiver: claim.paymentReceiver,
+      cost: claimParameters.cost,
+      erc20: claim.erc20
+    });
+    emit GachaClaimUpdated(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaim}.
+   */
+  function getClaim(address creatorContractAddress, uint256 instanceId) public view override returns (Claim memory) {
+    return _getClaim(creatorContractAddress, instanceId);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-getClaimForToken}.
+   */
+  function getClaimForToken(
+    address creatorContractAddress,
+    uint256 tokenId
+  ) external view override returns (uint256 instanceId, Claim memory claim) {
+    instanceId = _tokenInstances[creatorContractAddress][tokenId];
+    claim = _getClaim(creatorContractAddress, instanceId);
+  }
+
+  function _getClaim(address creatorContractAddress, uint256 instanceId) private view returns (Claim storage claim) {
+    claim = _claims[creatorContractAddress][instanceId];
+    if (claim.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.ClaimNotInitialized();
+  }
+
+  /**
+   * See {IGachaLazyClaim-mintReserve}.
+   */
+  function mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) external payable override {
+    require(!Address.isContract(msg.sender), "Only EOAs can reserve a mint.");
+    _mintReserve(creatorContractAddress, instanceId, mintCount);
+  }
+
+  function _mintReserve(address creatorContractAddress, uint256 instanceId, uint32 mintCount) private {
+    Claim storage claim = _getClaim(creatorContractAddress, instanceId);
+    // Checks for reserving
+    if (claim.startDate > block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (claim.endDate > 0 && claim.endDate <= block.timestamp) revert IGachaLazyClaim.ClaimInactive();
+    if (msg.value < (claim.cost + MINT_FEE) * mintCount) revert IGachaLazyClaim.InvalidPayment();
+    uint32 amountAvailable = claim.totalMax - claim.total;
+    if (amountAvailable == 0 ) revert IGachaLazyClaim.ClaimSoldOut();
+
+    // calculate the amount to reserve and update totals
+    uint32 amountToReserve = uint32(Math.min(mintCount, amountAvailable));
+    claim.total += amountToReserve;
+    _reservedMintsPerWallet[creatorContractAddress][instanceId][msg.sender] += amountToReserve;
+
+    // Refund any overpayment
+    if (msg.value > (claim.cost + MINT_FEE) * amountToReserve){
```

Comment:
There is a far cheaper check.  Every time you do additions and multiplications, you increase gas usage
Because we should be enforcing equality of msg.value above, all you need to check here is
if (amountToReserve != mintCount)